src/Symfony/Component/DependencyInjection/ParameterBag/ContainerBagInterface.php

```diff
@@ -29,6 +29,10 @@ public function all(): array;
     /**
      * Replaces parameter placeholders (%name%) by their values.
      *
+     * @template TValue
+     * @param TValue $value
+     * @return TValue
```

Comment:
This is wrong. If you pass a string as input and that string contains only a placeholder, it could change its type (as it will return the value of the parameter)

Reply:
Hmm, but we are using it as if it has the contract `resolveValue(array<array>): array<array>` in https://github.com/symfony/symfony/blob/9ccf3651a8a2ac7aee0002f67e8a27b2b85d34d0/src/Symfony/Component/DependencyInjection/Compiler/MergeExtensionConfigurationPass.php#L62-L77

Reply:
Well, if you pass an array in, you always get an array on the output (but passing an array of strings does not guarantee that the returned array also contains strings). But if you pass a string in, the output can be anything (well, a scalar, null or an array of those as we don't allow objects or resources in parameters)

Reply:
Ok, I've tried to convey this more precise contract using a conditional return type.